### ISSUE TITLE
How long will training process take with Chinese sentence generation

### ISSUE THREAD

**Issue by wenbotse** [Tony607/Yelp_review_generation#1]
Hello, i download your code and modify little for Chinese sentence generation, while the loss reduce very slow and keep on a large value and don't reduce. Should I be more patience about the training process or maybe the code i modified was wrong. May I try deeper layers?
mycode : https://github.com/wenbotse/char_level_generator/blob/master/zh_train.py
training data like this :
《SOR》 农家饭做的很好吃，价格不贵。住宿条件很干净，老板特别热情，每年我都会带的家人朋友去一次，老板还给我们提供景区免费票。最后结账的时候会给打折。《EOR》
thanks for your reply .

**Comment by Tony607** [comment]
Hi Wenbo,
In my English example, there are only 95 unique characters. In Chinese,  are you doing the same for all unique Chinese characters? There might be too much if that is the case. Maybe try to turn Chinese Characters to PinYin first and train on that. Then for the inference, use a dictionary checking algorithm to reverse PinYin back to the most likely Chinese characters. That is just my thought. If you going to try it, let me know the result.

**Comment by wenbotse** [comment]
Hi Tony,
  I have try the code like this : https://github.com/wenbotse/char_level_generator/blob/master/zh_deep_dot_train.py
For Chinese sentence ，i choose a common char set no more than 1000 char, and add some lstm layers and dropout layers,because the target space is much larger than english sentence, so we need more hidden layers and more hidden units . 
some results like this ,while i think it's a little overfit.
I want to try some more layers & hidden units while don't have so much computation resources.....


<SOR>味道很不错的一家店，环境一般。<EOR>
<SOR>在这里聚会是个不错的选择。<EOR>
<SOR>环境真的要赞一个，非常非常的好。<EOR>
<SOR>服务很好， 停车太不好听了。<EOR>
<SOR>喜欢的菜很多<EOR>
<SOR>挺喜欢的<EOR>
<SOR>聚会达人<EOR>
<SOR>过油肉很香很好吃，每次必点<EOR>
<SOR>来这里还是推荐老北京的小吃，一些热炒的口味实在一般。<EOR>
<SOR>老店了。但是服务些热炒的口味实在一般。<EOR>

**Comment by Tony607** [comment]
Hi Wenbo,
I see in your code have stacked 5 LSTM layers, that is too much. 2 Stacked LSTMs are already quite computationally expensive.
Maybe try the following,
1. Shorter training sentences
2. Re-exam the training dataset and only keep "5-start" reviews.
3. Try use GRU to replace all LSTMs. GRU is simpler and can be trained faster.
